plugin.d.ts

```diff
@@ -54,20 +54,26 @@ export default function fp<
   options?: string
 ): FastifyPluginCallback<Options>;
 
-export interface PluginMetadata {
-  /** Bare-minimum version of Fastify for your plugin, just add the semver range that you need. */
-  fastify?: string,
-  name?: string,
-  /** Decorator dependencies for this plugin */
-  decorators?: {
-    fastify?: (string | symbol)[],
-    reply?: (string | symbol)[],
-    request?: (string | symbol)[]
-  },
-  /** The plugin dependencies */
-  dependencies?: string[],
-  encapsulate?: boolean
+declare namespace fp {
+
+  export interface PluginMetadata {
+    /** Bare-minimum version of Fastify for your plugin, just add the semver range that you need. */
+    fastify?: string,
+    name?: string,
+    /** Decorator dependencies for this plugin */
+    decorators?: {
+      fastify?: (string | symbol)[],
+      reply?: (string | symbol)[],
+      request?: (string | symbol)[]
+    },
+    /** The plugin dependencies */
+    dependencies?: string[],
+    encapsulate?: boolean
+  }
+
+  // Exporting PluginOptions for backward compatibility after renaming it to PluginMetadata
+  export interface PluginOptions extends PluginMetadata {}
+
```

Comment:
```suggestion

  // default export for ESM
  export { fp as default } 
```

Reply:
Oops yes this was required in my proof of concept but missed it here. Thanks for spotting it!!